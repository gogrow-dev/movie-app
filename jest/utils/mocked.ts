import { Category, Movie } from '../../src/store/actions/moviesActions';

export const mockedCategories: Category[] = [
  {
    id: 1,
    name: 'Category 1',
  },
  {
    id: 2,
    name: 'Category 2',
  },
];

export const mockedMovies: Movie[] = [
  {
    id: 1,
    title: 'Title',
    releaseDate: '2022-01-01',
    posterPath: '/path.png',
    voteAverage: 8.1,
    voteCount: 1234,
    backdropPath: '/path.png',
    genreIds: [],
    overview: 'Movie Overview',
    popularity: 10,
    bookmark: false,
  },
  {
    id: 2,
    title: 'Title 2',
    releaseDate: '2022-01-01',
    posterPath: '/path.png',
    voteAverage: 8.1,
    voteCount: 1234,
    backdropPath: '/path.png',
    genreIds: [],
    overview: 'Movie Overview',
    popularity: 10,
    bookmark: false,
  },
];
