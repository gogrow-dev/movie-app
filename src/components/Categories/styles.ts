import { StyleSheet } from 'react-native';
import colors from '../../styles/colors';

export default StyleSheet.create({
  container: {
    marginTop: 22,
  },
  list: {
    marginTop: 18,
  },
  listContent: {
    paddingHorizontal: 29,
  },
  title: {
    marginLeft: 29,
    fontFamily: 'Poppins-Bold',
    fontSize: 18,
    color: colors.white,
  },
});
