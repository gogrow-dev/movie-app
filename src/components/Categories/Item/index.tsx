import { View, Text } from 'react-native';
import React from 'react';
import styles from './styles';

interface IItem {
  name: string;
}

const Item = ({ name }: IItem) => {
  return (
    <View style={[styles.container, styles.containerSelected]}>
      <Text style={[styles.text, styles.textSelected]}>{name}</Text>
    </View>
  );
};

export default Item;
