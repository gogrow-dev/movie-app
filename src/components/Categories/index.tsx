import { View, Text, FlatList } from 'react-native';
import React from 'react';
import styles from './styles';
import Item from './Item';

const data = ['Top Rated', 'Popular', 'Action'];

const Categories = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Categories</Text>
      <FlatList<string>
        showsHorizontalScrollIndicator={false}
        style={styles.list}
        contentContainerStyle={styles.listContent}
        horizontal
        data={data}
        keyExtractor={(_, index) => `category-${index}`}
        renderItem={({ item }) => <Item name={item} />}
      />
    </View>
  );
};

export default Categories;
