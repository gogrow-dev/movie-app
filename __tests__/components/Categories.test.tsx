import 'react-native';
import React from 'react';
import * as redux from 'react-redux';
import renderer from 'react-test-renderer';
import Categories from '../../src/components/Categories';
import { mockedCategories } from '../../jest/utils/mocked';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

jest.spyOn(redux, 'useDispatch').mockImplementation(() => jest.fn());
jest.spyOn(redux, 'useSelector').mockImplementation(() => mockedCategories);

describe('Testing MovieCard', () => {
  it('Create snapshot', async () => {
    const tree = await renderer.create(<Categories />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
