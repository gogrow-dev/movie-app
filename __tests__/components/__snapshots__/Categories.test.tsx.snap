// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieCard Create snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 22,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#EEEEEE",
        "fontFamily": "Poppins-Bold",
        "fontSize": 18,
        "marginLeft": 29,
      }
    }
  >
    Categories
  </Text>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingHorizontal": 29,
      }
    }
    data={
      Array [
        Object {
          "id": 1,
          "name": "Category 1",
        },
        Object {
          "id": 2,
          "name": "Category 2",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginTop": 18,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#3A3F47",
                  "borderRadius": 16,
                  "height": 32,
                  "justifyContent": "center",
                  "marginRight": 12,
                  "paddingHorizontal": 22,
                },
                false,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#EEEEEE",
                    "fontFamily": "Poppins-SemiBold",
                    "fontSize": 16,
                  },
                  false,
                ]
              }
            >
              Category 1
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#3A3F47",
                  "borderRadius": 16,
                  "height": 32,
                  "justifyContent": "center",
                  "marginRight": 12,
                  "paddingHorizontal": 22,
                },
                false,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#EEEEEE",
                    "fontFamily": "Poppins-SemiBold",
                    "fontSize": 16,
                  },
                  false,
                ]
              }
            >
              Category 2
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
